tensorbay/opendataset/COCO2017/loader.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+import json
+import os
+from collections import defaultdict
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+
+from ...dataset import Data, Dataset
+from ...label import (
+    Label,
+    LabeledBox2D,
+    LabeledKeypoints2D,
+    LabeledMultiPolygon,
+    LabeledRLE,
+    PanopticMask,
+)
+from ...utility.itertools import chunked
+from .._utility.glob import glob
+
+try:
+    from PIL import Image
+except ModuleNotFoundError:
+    from .._utility.mocker import Image
+
+DATASET_NAME = "COCO2017"
+_SEGMENT_NAMES = ("train", "val", "test", "unlabeled")
+
+
+def COCO2017(path: str) -> Dataset:
+    """Dataloader of COCO2017 open dataset.
+
+    .. _COCO2017: https://cocodataset.org/#home
+
+    The file structure should be like::
+
+        <path>
+            annotations/
+                panoptic_train2017/
+                    000000116037.png
+                    000000116040.png
+                    ...
+
+                panoptic_val2017/
+                test_panoptic_val2017/
+                instances_train2017.json
+                instances_val2017.json
+                panoptic_train2017.json
+                panoptic_val2017.json
+                person_keypoints_train2017.json
+                person_keypoints_val2017.json
+
+            train2017/
+                000000116026.jpg
+                000000116031.jpg
+
+            test2017/
+            val2017/
+            unlabeled2017/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENT_NAMES:
+        segment = dataset.create_segment(segment_name)
+        if segment_name in {"test", "unlabeled"}:
+            for image_path in glob(os.path.join(root_path, f"{segment_name}2017", "*.jpg")):
+                segment.append(Data(image_path))
+        else:
+            annotation_path = os.path.join(root_path, "annotations")
+            task_information = _get_annotations(annotation_path, segment_name)
+            for image_path in glob(os.path.join(root_path, f"{segment_name}2017", "*.jpg")):
+                data = Data(image_path)
+
+                image_name = os.path.splitext(os.path.basename(image_path))[0]
```

Comment:
we have a word to describe a filename without extension, which is `stem`, you can find it in python `pathlib`